### PR TITLE
Update error message for `mesheryctl system start`

### LINKED ISSUE #6267: [mesheryctl] improve the error messages shown to users for `mesheryctl system start`
_See [mesheryctl Command Tracker](https://bit.ly/3dqXy1q) for current status of commands._ For additional context, see https://github.com/meshery/meshery/pull/5661.

#### Current Behavior
Responses from `mesheryctl` commands don't always have the best user experience in mind. 

#### Desired Behavior
Improve the error messages in mesheryctl commands from the user's perspective. 
1. Ensure that there are proper error messages when users:
    1. Give more arguments to a command than required
    1. Misspell a command
    1. Do not give the required input via flags or args
    1. Use the command without any arguments or flags.
1. The error messages being displayed to users are meaningful from user's perspective. And if applicable, a potential fix for the problem should be shown too.  For example
    1. "failed to unmarshal response body" could be modified to "couldn't process JSON response from server"
    1. "manifest not found" could be modified to "manifest not found. Please provide the path to the app manifest."

#### Implementation
The commands in scope of this issue are:

- [ ] `mesheryctl system start`
- [ ] `mesheryctl system start --skip-update`
- [ ] `mesheryctl system start --skip-browser`
- [ ] `mesheryctl system start --skip-reset`
- [ ] `mesheryctl system start --platform`

#### Acceptance Tests
1. Contributor adhered to the [Meshery CLI Style Guide](https://docs.meshery.io/project/contributing/contributing-cli#meshery-cli-style-guide).

 --- 
  #### Contributor Resources

- ⌨️ [mesheryctl Contributing Guide](https://github.com/meshery/meshery/blob/master/mesheryctl/README.md)
- [Beginner's guide to contributing to Meshery and mesheryctl](https://youtu.be/hh_kFLZx3G4)
- [mesheryctl Command Tracker](https://docs.google.com/spreadsheets/d/1q63sIGAuCnIeDs8PeM-0BAkNj8BBgPUXhLbe1Y-318o/edit#gid=0)
- [Meshery CLI Commands and Documentation](https://docs.google.com/document/d/1xRlFpElRmybJ3WacgPKXgCSiQ2poJl3iCCV1dAalf0k/edit#heading=h.5fucij4hc5wt)
- 🙋🏾🙋🏼 Questions: [Layer5 Discussion Forum](https://discuss.layer5.io) and [Layer5 Community Slack](http://slack.layer5.io)
  📚 For the **Reference** head over to this [pr](https://github.com/meshery/meshery/pull/5661).

### PR DESCRIPTION
**Pull request by rakshitgondwal**
**Notes for Reviewers**
- Updated the error message when the user provides some other platform in `mesheryctl system start --platform`.
- Updated the error message `mesheryctl system start --skip-reset`

This PR fixes #6267 

**[Signed commits](https://github.com/meshery/meshery/blob/master/CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**
- [x] Yes, I signed my commits.

### CODE REVIEW

**Comment by rakshitgondwal** [comment]
@leecalcote 
- I could not locate the use `--skip-browser` flag function in `start.go`.
- I made an observation that if a user gives the command `mesheryctl system start --platform` without giving an args to the command, it just shows an error stating that `Err: flag needs an argument`. I tried to solve this by creating a new function inside of `system/error.go`, but I was getting an error, so I'll need a little of guidance over there.
- If a user enters a command say `mesheryctl system start --platform dokcer` and the user has misspelled the word `docker` as `dokcer`, then rather then showing him/her the command unavailable error, we can show him a misspell command error. What do you think?

**Comment by leecalcote** [comment]
@gopivaibhav it'd be good of you to offer review.

File: mesheryctl/internal/cli/root/system/error.go [hunk 1]
```diff
@@ -67,7 +67,7 @@ func ErrUnmarshal(err error, obj string) error {
 }
 
 func ErrUnsupportedPlatform(platform string, config string) error {
-	return errors.New(ErrUnsupportedPlatformCode, errors.Alert, []string{}, []string{"the platform ", platform, " is not supported. Supported platforms are:\n\n- docker\n- kubernetes\n\nVerify this setting in your meshconfig at ", config, " or verify by executing `mesheryctl system context view`"}, []string{}, []string{})
+	return errors.New(ErrUnsupportedPlatformCode, errors.Alert, []string{}, []string{"The platform ", platform, " is not supported for the deployement of Meshery. Supported platforms are:\n\n- docker\n- kubernetes\n\nVerify this setting in your meshconfig at ", config, " or verify by executing `mesheryctl system context view`"}, []string{}, []string{})
```
Comment (alphaX86) [inline mesheryctl/internal/cli/root/system/error.go:70]:
```suggestion
	return errors.New(ErrUnsupportedPlatformCode, errors.Alert, []string{}, []string{"The platform ", platform, " is not supported for the deployment of Meshery. Supported platforms are:\n\n- docker\n- kubernetes\n\nVerify this setting in your meshconfig at ", config, " or verify by executing `mesheryctl system context view`"}, []string{}, []string{})
```

File: mesheryctl/internal/cli/root/system/error.go [hunk 2]
```diff
@@ -47,7 +47,7 @@ func ErrStopMeshery(err error) error {
 }
 
 func ErrResetMeshconfig(err error) error {
-	return errors.New(ErrResetMeshconfigCode, errors.Alert, []string{"Error resetting meshconfig"}, []string{err.Error()}, []string{"Meshery server config file is not reset to default settings"}, []string{"Verify Meshery server config file is reset to default settings by executing `mesheryctl system context view`"})
+	return errors.New(ErrResetMeshconfigCode, errors.Alert, []string{"Error resetting meshconfig to default settings"}, []string{err.Error()}, []string{"Meshery server config file is not reset to default settings"}, []string{"Verify Meshery server config file is reset to default settings by executing `mesheryctl system context view`"})
```
Comment (alphaX86) [inline mesheryctl/internal/cli/root/system/error.go:50]: This one seems ok to me...

**Review by alphaX86 — CHANGES_REQUESTED** [review]

File: mesheryctl/internal/cli/root/system/error.go (hunk 1 above)
Comment (rakshitgondwal) [reply mesheryctl/internal/cli/root/system/error.go]: Done!

**Comment by rakshitgondwal** [comment]
> @leecalcote
> 
> * I could not locate the use `--skip-browser` flag function in `start.go`.
> * I made an observation that if a user gives the command `mesheryctl system start --platform` without giving an args to the command, it just shows an error stating that `Err: flag needs an argument`. I tried to solve this by creating a new function inside of `system/error.go`, but I was getting an error, so I'll need a little of guidance over there.
> * If a user enters a command say `mesheryctl system start --platform dokcer` and the user has misspelled the word `docker` as `dokcer`, then rather then showing him/her the command unavailable error, we can show him a misspell command error. What do you think?

Any views on this? @alphaX86

**Comment by leecalcote** [comment]
> If a user enters a command say mesheryctl system start --platform dokcer and the user has misspelled the word docker as dokcer, then rather then showing him/her the command unavailable error, we can show him a misspell command error. What do you think?

Through use of a library, ideally, one with first-class integration into cobra, yes (as a separate issue to open, solution to research, and proposal to make).

**Comment by leecalcote** [comment]
@abdullah1308 will you lay your eyes on this PR and give it a review?

**Comment by abdullah1308** [comment]
> 
> * I could not locate the use `--skip-browser` flag function in `start.go`.
> * I made an observation that if a user gives the command `mesheryctl system start --platform` without giving an args to the command, it just shows an error stating that `Err: flag needs an argument`. I tried to solve this by creating a new function inside of `system/error.go`, but I was getting an error, so I'll need a little of guidance over there.
> * If a user enters a command say `mesheryctl system start --platform dokcer` and the user has misspelled the word `docker` as `dokcer`, then rather then showing him/her the command unavailable error, we can show him a misspell command error. What do you think?

@rakshitgondwal 

* It is required that you sign your commits when contributing to Meshery as per the [contributing guidelines](https://github.com/meshery/meshery/blob/master/CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin). You may refer [here](https://stackoverflow.com/questions/13043357/git-sign-off-previous-commits) to sign previous commits.

* The `--skip-browser` flag is implemented here 
https://github.com/meshery/meshery/blob/d601d13acb3793f77592cdba5175528c56e4f758/mesheryctl/internal/cli/root/system/dashboard.go#L204-L212

* The message `Error: flag needs an argument: --platform` is returned by the cobra library and is self explanatory. You need not override it.

**Comment by rakshitgondwal** [comment]
Thank you for the review! @abdullah1308 
I'll look for the changes in `--skip-browser` asap. Also, I did sign my commits. Please let me know if I did it in the wrong way or is there any other issue.

**Comment by abdullah1308** [comment]
> Thank you for the review! @abdullah1308 I'll look for the changes in `--skip-browser` asap. Also, I did sign my commits. Please let me know if I did it in the wrong way or is there any other issue.

@rakshitgondwal Did you use the `-s`  or `--signoff` flag to do so ? I think you haven't. Your commit messages have the following sign-off -

`Signed-off-by: Rakshit Gondwal rakshitgondwal3@gmail.com`

This sign-off would pass the DCO check -

`Signed-off-by: Rakshit Gondwal <rakshitgondwal3@gmail.com>`

**Comment by leecalcote** [comment]
Adding to @abdullah1308's excellent notes:

In your local branch:

```
git rebase HEAD~3 --signoff
git push --force-with-lease origin dev1
```

**Comment by rakshitgondwal** [comment]
Thanks a lot for the update @leecalcote @abdullah1308. I am so sorry, I am not being able to work on this as I'm travelling right now. I'll complete this as soon as I reach home.

**Comment by rakshitgondwal** [comment]
@leecalcote Sorry for the tag but can you please review this?

**Review by leecalcote — APPROVED** [review]

**Comment by leecalcote** [comment]
Three verbiage improvements approved. Those look good, @rakshitgondwal 👍

**Comment by rakshitgondwal** [comment]
Thanks a lot for the merge! @leecalcote

**Pull request merged**